src/lab16.py

```diff
@@ -252,14 +259,37 @@ def layout(self):
             child.y = baseline + child.get_ascent()
         max_descent = max([child.get_descent(1.25)
                            for child in self.children])
-        self.height = max_ascent + max_descent
-        self.parent.dirty_height = True
-        mark_dirty(self.parent)
 
-        self.dirty_width = False
-        self.dirty_height = False
-        self.dirty_x = False
-        self.dirty_y = False
+        self.height = self.height_field.set(max_ascent + max_descent)
+        
+class DependentField:
```

Comment:
Two main comments about this abstraction:

* It's common to simply set a field based on the field of the thing it depends on. I suggest making a version of set that does this automatically, without first setting a local variable. The other common pattern is adding together two depended-on values (and maybe that's all the cases?). Also make a version for adding two things?

* It's weird that there are values and dependent fields that represent that value, and duplicate the value itself. 
e.g. there is `zoom` and `zoom_field`. Why do we need both?

Reply:
Point one makes sense. I can introduce those helpers.